### ISSUE TITLE
(MEDIUM) Displayed precision (no. of decimal places) is different  for fees and for amount to be minted

### ISSUE THREAD

**Issue by michalinacienciala** [threshold-network/token-dashboard#371]
When the `Initiate minting tBTC` pop-up is displayed to the user in Step 3, it shows the values of the tBTC that will be minted and the fees that will be subtracted. The fees are displayed with the precision of 6 decimal places, but it looks to me that the minted amount is displayed with the precision of two decimal places (and rounded down). So despite I sent 0.016 BTC to the deposit address, I can see only 0.01 as a minted amount. As a user, knowing that the fees are presented with the 6 digit precision, I would assume that the minted amount is also presented with the same precision and would be wondering where my ~0.006 BTC has gone.
My suggestion would be to show the same 6 digit precision for the fees and for the minted amount. For the value in the title I think we could leave the two decimal points.

Wondering what's @sashatanase's opinion here.
Also, how should we round up the value in case of `0.099999` minted amount - to `0.10` or `0.09`? Mathematical standard is to round fractions to the nearest number - which would be `0.10`. It looks to me that we always round the deposited amount to the lower value. Maybe there's some preference here from the point of UX?

![image](https://user-images.githubusercontent.com/78352137/214949327-bd6f0b5c-d19b-4350-ae8d-dca21874a997.png)

**Comment by pdyraga** [comment]
@michalinacienciala What do you think about promoting this issue to MEDIUM level? People can be confused/panic when they see fees are not adding up due to rounding in the dApp.

**Comment by michalinacienciala** [comment]
Agree, I was battling with myself when deciding between MINOR/MEDIUM :)

**Comment by michalinacienciala** [comment]
Generally what we want to achieve here is to not confuse a user about the displayed values.

Current design may be confusing because:
- user sees some of the values rounded to two decimal places, some to six
- the values seems to be always rounded down (i.e. `0.019` becomes `0.01`, `0.00000855` becomes `0.00008`), which is not the usual standard for rounding (we should round to the nearest value, i.e. (`0.019` becomes `0.02`, `0.00000855` becomes `0.00009`)

There are couple of ways we can tackle the problem here.
a) we can show the values with full precision (no confusion, but also hard to read)
b) we can show the `Minted Amount`, `Bitcoin Miner Fee` and `Threshold Network Fee` with 6 digit precision, and leave the amount in the title with 2 digit precision; in all those places use the standard rounding rules
c) same as b), but in the title show `~` char before the value (i.e. `0.019` becomes `~0.02`) 
d) same as b), but in the title show `<` / `>` char before the value (i.e. `0.019` becomes `<0.02`) and display value with higher precision after hovering on that number
e) we add `<` / `>` + full value after hovering in more places
f) various combinations of above

To decide which solution to go for at the moment, we need to assess how hard to implement they are.
We can do some easier solution now and improve on it after the Launch. We just need to make sure user isn't confused by the presented values.

**Comment by michalinacienciala** [comment]
We  had a discussion with @pdyraga, @r-czajkowski and @michalsmiarowski about this. We need to implemet this in a way that s not confusing for the user and is not too time consuming. We may do some improvements on that later.
@r-czajkowski will likely implement a solution which behaves like this:
`0,001` -> `<0.01`
`0.016` -> `~0.02`
and amounts with higher precision will be displayed after hovering on the rounded values.
The fee/minted amount values in the `Initiate minting tBTC` screen and `Step 4` (except the title ones) will be written with 6 digits precision. In rest of the places we'll display 2 decimal places.

**Comment by michalinacienciala** [comment]
Btw, @liz-shinn had following situation:


![Image](https://user-images.githubusercontent.com/78352137/215123031-7f941d54-9e8d-4830-84bb-35359c0870b1.png)

I thought that when we have only one tx in the tx history, the amount in tx history and balance would be the same, but it looks that we have some difference there when it comes to the rounding mechanism (or use different source data). Please make sure that the displayed values in such case will be the same.

**Comment by michalinacienciala** [comment]
I'm reopening the issue.
I discovered that due to the fact that we now round amount to be minted to 6 decimal digits and fees to 8 digits in dApp, the user may get slight difference between the amount of deposited BTC and amount of minted tBTC plus fees. The difference is not big (no more than 0,00000050 BTC, I think,  which is worth ~ 1 cent USD) but may be a bit confusing for the user.
For example users depositing 0.015 BTC see in dApp that 0,01500049 got minted (their tBTC + fees).
```
                    chain         dApp
0 deposited BTC     0,0150000000  0,015
﻿
1 depositor`s TBTC  0,0149625150  0,014963
2 treasury fee      0,0000075000  0,0000075
3 minting fee       0,0000299850  0,00002999
﻿
4 sum 1+2+3         0,0150000000  0,01500049
5 diff 4-0          0             0,00000049
```

Matt:
> Good practice: be consistent with numeric presentation for any numbers that are meant to be compared with one another, eg if they're all in the same table in an interface.

So we should update the view (the `Initiate minting tBTC` pop-up and `Success / Step 4` page`) to either show all the values with 6 digits accuracy on hover, or 8 digits.
